pkg/controller/schedule_controller.go

```diff
@@ -127,10 +128,14 @@ func (c *scheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	// check for the schedule being due to run, and submit a Backup if so.
+	// Check for the schedule being due to run.
+	// If there are backup created by this schedule still in New or InProgress state,
+	// skip current backup creation to avoid running overlap backups.
 	// As the schedule must be validated before checking whether it's due, we cannot put the checking log in Predicate
-	if err := c.submitBackupIfDue(ctx, schedule, cronSchedule); err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "error running submitBackupIfDue for schedule %s", req.String())
+	if c.ifDue(schedule, cronSchedule) && !c.checkIfBackupInNewOrProgress(schedule) {
+		if err := c.submitBackup(ctx, schedule); err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "error submit backup for schedule %s", req.String())
+		}
```

Comment:
How about adding a log to explain why the schedule doesn't trigger a new backup?

Reply:
Added in function checkIfBackupInNewOrProgress